### DISCUSSION TITLE
Run on ESP32 / Open Book 2023

### DISCUSSION THREAD

**Discussion by ryanwwest** [koreader/koreader#11004] [Ideas]
I just saw a video about a cool open-source hardware ereader called Open Book (https://www.youtube.com/watch?v=vFD9V8Hh7Yg). It's a neat concept designed to educate its users as they build it while staying cheap. https://hackaday.io/project/192688-the-open-book

However, the ereader uses an ESP32-S3 module and has [this firmware](https://github.com/joeycastillo/libros). I'm not familiar with it but was wondering if there would be any shot at getting something like KOReader to run on it, or if it's too intense of an app / requires more of an OS? Another challenge is I think it lacks a touch screen.

**Comment by Frenzie** [comment]
I doubt KOReader is realistically usable on a device with less than about 128 MB of RAM, but perhaps the koreader-base repo could still be of use to someone. crengine was originally created for such limited devices.

**Reply by ryanwwest** [reply]
128MB is a good benchmark to be aware of, thanks. So the [Raspberry Pi Zero 2 W](https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/) could do the job just fine with Linux, but maybe not much less than that. Maybe there are less powerful boards I'm not aware that could do it.
